### PR TITLE
Remove redundant rule validations in App Handler

### PR DESCRIPTION
**Pull request by TJ-Everett**

### CODE REVIEW

File: src/client/application/ProtocolApplicationHandler.sol [hunk 1]
```diff
@@ -368,7 +366,6 @@ contract ProtocolApplicationHandler is
      */
     // slither-disable-next-line calls-loop
     function setAccountMaxValueByRiskScoreIdUpdate(ActionTypes _action, uint32 _ruleId) internal {
-        IRuleProcessor(ruleProcessor).validateAccountMaxValueByRiskScore(createActionTypesArray(_action), _ruleId);
```
Comment (Palmerg4) [inline src/client/application/ProtocolApplicationHandler.sol:374]: Is this supposed to be removed? I think this is why it is redundant to have the check before calling this method. Maybe I'm misunderstanding.

Comment (TJ-Everett) [reply src/client/application/ProtocolApplicationHandler.sol:374]: You are right that this is the call. The other functions have checks prior to calling this one. It looked like this was how we were doing it in the setAccountMaxValueByAccessLevelId and accountDenyForNoAccess so I made the call to remove this and leave the validations in the calling function. 

I could be persuaded to remove the other two and keep the validation in the internal function. 

Am I understanding that correctly?

Comment (Palmerg4) [reply src/client/application/ProtocolApplicationHandler.sol:374]: Ahh that makes sense. I think either way is fine, no need to keep the validation in the internal function. But it does look like the validations in the calling functions are removed as well. That's where the confusion is coming from

Comment (ShaneDuncan602) [reply src/client/application/ProtocolApplicationHandler.sol:374]: IMO, the cleanest way to do these validations would be within the internal function. That way, there are no repetitious calls and less likelihood of them in the future. I could see us creating alternate entry points for changing rules. If so, having the validations in the internal function could be advantageous.

Comment (TJ-Everett) [reply src/client/application/ProtocolApplicationHandler.sol:374]: Good call guys, I think I was even on team internal function last time this popped up. I have all the validation calls dropped into the internal functions now.

File: src/client/application/ProtocolApplicationHandler.sol [hunk 2]
```diff
@@ -341,11 +340,8 @@ contract ProtocolApplicationHandler is
      * @param _ruleIds Rule Id corresponding to the actions
      */
     function setAccountMaxValueByRiskScoreIdFull(ActionTypes[] calldata _actions, uint32[] calldata _ruleIds) external ruleAdministratorOnly(appManagerAddress) {
-        validateRuleInputFull(_actions, _ruleIds);
```
Comment (Palmerg4) [inline src/client/application/ProtocolApplicationHandler.sol:344]: Do we want to get rid of the length checks?

File: src/client/application/ProtocolApplicationHandler.sol [hunk 3]
```diff
@@ -502,11 +497,8 @@ contract ProtocolApplicationHandler is
      * @param _ruleIds Rule Id corresponding to the actions
      */
     function setAccountMaxValueByAccessLevelIdFull(ActionTypes[] calldata _actions, uint32[] calldata _ruleIds) external ruleAdministratorOnly(appManagerAddress) {
-        validateRuleInputFull(_actions, _ruleIds);
```
Comment (Palmerg4) [inline src/client/application/ProtocolApplicationHandler.sol:505]: Here as well, I think we need to keep the length checks here

File: src/client/application/ProtocolApplicationHandler.sol [hunk 4]
```diff
@@ -588,11 +581,8 @@ contract ProtocolApplicationHandler is
      * @param _ruleIds Rule Id corresponding to the actions
      */
     function setAccountMaxValueOutByAccessLevelIdFull(ActionTypes[] calldata _actions, uint32[] calldata _ruleIds) external ruleAdministratorOnly(appManagerAddress) {
-        validateRuleInputFull(_actions, _ruleIds);
```
Comment (Palmerg4) [inline src/client/application/ProtocolApplicationHandler.sol:591]: Same as previous comments, length check

File: src/client/application/ProtocolApplicationHandler.sol [hunk 5]
```diff
@@ -674,11 +665,8 @@ contract ProtocolApplicationHandler is
      * @param _ruleIds Rule Id corresponding to the actions
      */
     function setAccountMaxTxValueByRiskScoreIdFull(ActionTypes[] calldata _actions, uint32[] calldata _ruleIds) external ruleAdministratorOnly(appManagerAddress) {
-        validateRuleInputFull(_actions, _ruleIds);
```
Comment (Palmerg4) [inline src/client/application/ProtocolApplicationHandler.sol:677]: Here as well

File: src/client/application/ProtocolApplicationHandler.sol (hunk 2 above)
Comment (TJ-Everett) [reply src/client/application/ProtocolApplicationHandler.sol:344]: great catch thank you. I think I got over zealous with the trimming!

**Review by Palmerg4 — APPROVED** [review]
LGTM!

File: docs/userGuides/README.md [hunk 6]
```diff
@@ -103,15 +103,15 @@ Be sure to [set environmental variables](./deployment/SET-ENVIRONMENT.md) and so
 `scripts/deploy/DeployProtocol.sh`
 This script is responsible for deploying all the protocol contracts. Take into account that no application-specific contracts are deployed here.
 
-#### Deploy Some Test Game Tokens
+#### Deploy Some Test Application Tokens
```
Comment (ShaneDuncan602) [inline docs/userGuides/README.md:106]: Holy cow. Can't believe we still had a game reference in here.

**Review by ShaneDuncan602 — APPROVED** [review]
Looks great.

**Pull request merged**